sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -436,57 +437,90 @@ static size_t _handle_req(coap_pkt_t *pdu, uint8_t *buf, size_t len,
     return pdu_len;
 }
 
+static int _request_matcher_default(gcoap_listener_t *listener,
+                                    const coap_resource_t **resource,
+                                    const coap_pkt_t *pdu)
+{
+    uint8_t uri[CONFIG_NANOCOAP_URI_MAX];
+    int ret = GCOAP_RESOURCE_MISMATCH;
+
+    if (coap_get_uri_path(pdu, uri) <= 0) {
+        return GCOAP_RESOURCE_NO_PATH;
```

Comment:
Maybe add
```
/* The Uri-Path options are longer than CONFIG_NANOCOAP_URI_MAX, and thus
 * do not match anything that could be found on this host. */
```

although given that now there are other matchers that might cope with other paths, it should more be


```
/* The Uri-Path options are longer than CONFIG_NANOCOAP_URI_MAX, and thus
 * do not match anything that could be found by this handler. */
```
and just return that this matcher didn't find anything.